### ISSUE TITLE
Pull tariff data (price per kw etc) and provide as sensor to calculate cost on Energy dash

### ISSUE THREAD

**Issue by user897943** [HandyHat/ha-hildebrandglow-dcc#24]
**Is your feature request related to a problem? Please describe.**
Not a problem, but it's a missing feature

**Describe the solution you'd like**
If the Bright app is able to show costs, this means tariff pricing is available. Please expose this as an entity so we can use the option "Use an entity with current price" when adding as source to the energy dash. Should pull electric and gas prices for obvious reasons.

**Describe alternatives you've considered**
Alternative is to type the value manually but this is cumbersome, if you change tariff your meter will self-update however if you forget to update HA to amend the pricing manually, the calculation will be incorrect. This ensures the prices are always current and the costs are always as accurate as possible.

**Additional context**
No need for this, it's an obvious improvement.

**Comment by kivster** [comment]
This would be a welcomed and useful enhancement.

**Comment by totalitarian** [comment]
Yes please add to this excellent addon

**Comment by ColinRobbins** [comment]
Pull request created!

**Comment by ColinRobbins** [comment]
> Trying now, glad you merged in some of my changes. Not sure HandyHat is still maintaining this anymore....

No, I'm note sure either.   I suggest we give it time - we have no idea what @HandyHat  personal circumstances are.
If in a few months there is still no activity, I'll look at breaking free from the fork so we can maintain it going forward.

**Comment by totalitarian** [comment]
Anyone know why my energy usage would show negative values?

![image](https://user-images.githubusercontent.com/2385696/137195342-a05a387f-39a6-4ffd-ba01-798eaf5344e3.png)

This is my sensor

![image](https://user-images.githubusercontent.com/2385696/137195375-3ff5826d-b461-4879-9b52-c2820debd54b.png)

**Comment by ColinRobbins** [comment]
@totalitarian, the data from the sensor appears to be OK, and what I'd expect.   So it *seems* the issues is in how the Energy integration is interpreting it.
Along the lines of the @user897943 response, have you double checked in the Configuration / Energy page that the sensor is added in the "Grid Consumption" section, and not accidentally in the "return to grid"
